cmd/root.go

```diff
@@ -38,25 +45,67 @@ func Execute() error {
 }
 
 func init() {
+	cobra.OnInitialize(initConfig)
+
 	rootCmd.PersistentFlags().StringVar(
-		&ServerURL,
-		"server-url",
-		"http://localhost:8080",
-		"base URL for a Rosetta server",
+		&configurationFile,
+		"configuration-file",
+		"",
+		`Configuration file that provides connection and test settings.
+If you would like to generate a starter configuration file (populated
+with the defaults), run rosetta-cli configuration:create.
+
+Any fields not populated in the configuration file will be populated with
+default values.`,
 	)
+	rootCmd.AddCommand(versionCmd)
 
-	rootCmd.AddCommand(checkCmd)
+	// Configuration Commands
+	rootCmd.AddCommand(configurationCreateCmd)
+	rootCmd.AddCommand(configurationValidateCmd)
+
+	// Check commands
+	rootCmd.AddCommand(checkDataCmd)
+	rootCmd.AddCommand(checkConstructionCmd)
+
+	// View Commands
 	rootCmd.AddCommand(viewBlockCmd)
 	rootCmd.AddCommand(viewAccountCmd)
 	rootCmd.AddCommand(viewNetworkCmd)
-	rootCmd.AddCommand(createConfigurationCmd)
-	rootCmd.AddCommand(versionCmd)
+
+	// Utils
+	rootCmd.AddCommand(utilsAsserterConfigurationCmd)
+}
+
+func initConfig() {
+	var err error
+	if len(configurationFile) == 0 {
+		Config = configuration.DefaultConfiguration()
+	} else {
+		Config, err = configuration.LoadConfiguration(configurationFile)
+	}
+	if err != nil {
+		log.Fatalf("%s: unable to load configuration", err.Error())
+	}
+}
+
+func ensureDataDirectoryExists() {
+	// If data directory is not specified, we use a temporary directory
+	// and delete its contents when execution is complete.
+	if len(Config.Data.DataDirectory) == 0 {
+		tmpDir, err := utils.CreateTempDir()
+		if err != nil {
+			log.Fatal(fmt.Errorf("%w: unable to create temporary directory", err))
```

Comment:
Nit: looks like we're using  `log.Fatal(fmt.Errorf(` vs `log.Fatalf` at line 88

Reply:
Good catch!